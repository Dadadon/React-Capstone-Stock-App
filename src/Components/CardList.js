import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import CardItem from "./CardItem";
import Header from "./Header";
import { RetrieveStocks } from "./redux/Stock/stock";

const CardList = () => {
    const stockItems = useSelector((state) => state.stocks.stocks);
    console.log(stockItems);
    const dispatch = useDispatch();
    useEffect(() => {
        if (stockItems.length === 0) { dispatch(RetrieveStocks());}
    }, [dispatch]);

    return (
        <>
        <Header title="Crypto" subtitle={`Top ${stockItems.length}`} />
        <h5 className="text-muted p-1">Rank by Popularity</h5>
        <div className="d-flex flex-wrap">
            { stockItems.map((stockItem) => (
                <CardItem   key={stockItem.id} id={stockItem.id} name={stockItem.name} rank={stockItem.rank} symbol={stockItem.symbol}/>
            ))}

        </div>
        </>
        
    )
}

export default CardList;