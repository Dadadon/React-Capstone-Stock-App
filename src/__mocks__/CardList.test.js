import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../Components/redux/configureStore';
import CardList from '../Components/Cardlist';

describe('Card List container component', () => {
    test('Crypto displays correctly', () => {
        const crypto = render(
    
          <Provider store={store}>
    
            <CardList />
    
          </Provider>,
    
        );
        expect(crypto).toMatchSnapshot();
      });
})