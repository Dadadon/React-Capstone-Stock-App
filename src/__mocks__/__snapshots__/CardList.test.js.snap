// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card List container component  Rockets correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid bg-dark p-5"
      >
        <h1
          class="text-white text-center"
        >
          Crypto
        </h1>
        <h5
          class="text-white text-center"
        >
          Top 0
        </h5>
      </div>
      <h5
        class="text-muted p-1"
      >
        Rank by Popularity
      </h5>
      <div
        class="d-flex flex-wrap"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid bg-dark p-5"
    >
      <h1
        class="text-white text-center"
      >
        Crypto
      </h1>
      <h5
        class="text-white text-center"
      >
        Top 0
      </h5>
    </div>
    <h5
      class="text-muted p-1"
    >
      Rank by Popularity
    </h5>
    <div
      class="d-flex flex-wrap"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card List container component Cryo correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid bg-dark p-5"
      >
        <h1
          class="text-white text-center"
        >
          Crypto
        </h1>
        <h5
          class="text-white text-center"
        >
          Top 0
        </h5>
      </div>
      <h5
        class="text-muted p-1"
      >
        Rank by Popularity
      </h5>
      <div
        class="d-flex flex-wrap"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid bg-dark p-5"
    >
      <h1
        class="text-white text-center"
      >
        Crypto
      </h1>
      <h5
        class="text-white text-center"
      >
        Top 0
      </h5>
    </div>
    <h5
      class="text-muted p-1"
    >
      Rank by Popularity
    </h5>
    <div
      class="d-flex flex-wrap"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card List container component Crypto displays correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid bg-dark p-5"
      >
        <h1
          class="text-white text-center"
        >
          Crypto
        </h1>
        <h5
          class="text-white text-center"
        >
          Top 0
        </h5>
      </div>
      <h5
        class="text-muted p-1"
      >
        Rank by Popularity
      </h5>
      <div
        class="d-flex flex-wrap"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid bg-dark p-5"
    >
      <h1
        class="text-white text-center"
      >
        Crypto
      </h1>
      <h5
        class="text-white text-center"
      >
        Top 0
      </h5>
    </div>
    <h5
      class="text-muted p-1"
    >
      Rank by Popularity
    </h5>
    <div
      class="d-flex flex-wrap"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card List container component Crypto isplays correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid bg-dark p-5"
      >
        <h1
          class="text-white text-center"
        >
          Crypto
        </h1>
        <h5
          class="text-white text-center"
        >
          Top 0
        </h5>
      </div>
      <h5
        class="text-muted p-1"
      >
        Rank by Popularity
      </h5>
      <div
        class="d-flex flex-wrap"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid bg-dark p-5"
    >
      <h1
        class="text-white text-center"
      >
        Crypto
      </h1>
      <h5
        class="text-white text-center"
      >
        Top 0
      </h5>
    </div>
    <h5
      class="text-muted p-1"
    >
      Rank by Popularity
    </h5>
    <div
      class="d-flex flex-wrap"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card List container component Missions Rockets correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid bg-dark p-5"
      >
        <h1
          class="text-white text-center"
        >
          Crypto
        </h1>
        <h5
          class="text-white text-center"
        >
          Top 0
        </h5>
      </div>
      <h5
        class="text-muted p-1"
      >
        Rank by Popularity
      </h5>
      <div
        class="d-flex flex-wrap"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid bg-dark p-5"
    >
      <h1
        class="text-white text-center"
      >
        Crypto
      </h1>
      <h5
        class="text-white text-center"
      >
        Top 0
      </h5>
    </div>
    <h5
      class="text-muted p-1"
    >
      Rank by Popularity
    </h5>
    <div
      class="d-flex flex-wrap"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Card List container component St correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container-fluid bg-dark p-5"
      >
        <h1
          class="text-white text-center"
        >
          Crypto
        </h1>
        <h5
          class="text-white text-center"
        >
          Top 0
        </h5>
      </div>
      <h5
        class="text-muted p-1"
      >
        Rank by Popularity
      </h5>
      <div
        class="d-flex flex-wrap"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="container-fluid bg-dark p-5"
    >
      <h1
        class="text-white text-center"
      >
        Crypto
      </h1>
      <h5
        class="text-white text-center"
      >
        Top 0
      </h5>
    </div>
    <h5
      class="text-muted p-1"
    >
      Rank by Popularity
    </h5>
    <div
      class="d-flex flex-wrap"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
